community/kernel/src/main/java/org/neo4j/storageengine/api/StorageEngine.java

```diff
@@ -113,6 +114,14 @@ void createCommands(
      */
     void prepareForRecoveryRequired();
 
+    /**
+     * As part of the simplified counts store. This will initialize the count store in the storage engine
+     * when doing recovery.
+     */
+    void initFromSnapshot( CountsSnapshot snapshot );
```

Comment:
Feels strange to have methods related to counts store exposed in storage engine API. Shouldn't it be unaware of things that are underneath it?